e2e-tests/tfm/cypress/e2e/journeys/utilisation-reports/generate-keying-sheet.spec.js

```diff
@@ -0,0 +1,146 @@
+  });
+
+  it('can generate keying data when only one fee record for a facility is at MATCH status but does not generate the fixed fee adjustment or principal balance adjustment', () => {
+    const matchingPayment = PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP).withAmount(100).build();
+    const firstFeeRecord = firstFeeRecordBuilder.withStatus(FEE_RECORD_STATUS.MATCH).withPayments([matchingPayment]).build();
+    const secondFeeRecord = secondFeeRecordBuilder.withStatus(FEE_RECORD_STATUS.TO_DO).build();
+    cy.task(NODE_TASKS.INSERT_FEE_RECORDS_INTO_DB, [firstFeeRecord, secondFeeRecord]);
+
+    cy.visit(`utilisation-reports/${REPORT_ID}`);
+    pages.utilisationReportPage.premiumPaymentsTab.generateKeyingDataButton().click();
+    pages.checkKeyingDataPage.generateKeyingSheetDataButton().click();
+
+    cy.url().should('eq', relative(`/utilisation-reports/${REPORT_ID}#keying-sheet`));
+
+    pages.utilisationReportPage.keyingSheetTab.keyingSheetTableRow(FIRST_FEE_RECORD_ID).should('contain', 'TO DO');
+    pages.utilisationReportPage.keyingSheetTab.keyingSheetTableRow(SECOND_FEE_RECORD_ID).should('not.exist');
+
+    pages.utilisationReportPage.keyingSheetTab.fixedFeeAdjustmentDecrease(FIRST_FEE_RECORD_ID).should('contain', '-');
+    pages.utilisationReportPage.keyingSheetTab.fixedFeeAdjustmentIncrease(FIRST_FEE_RECORD_ID).should('contain', '-');
+
+    pages.utilisationReportPage.keyingSheetTab.principalBalanceAdjustmentDecrease(FIRST_FEE_RECORD_ID).should('contain', '-');
+    pages.utilisationReportPage.keyingSheetTab.principalBalanceAdjustmentIncrease(FIRST_FEE_RECORD_ID).should('contain', '-');
+  });
+
+  it('calculates the fixed fee adjustment and principal balance adjustment when all fee records for a facility are at MATCH status for only one of the fee records for the facility', () => {
+    const matchingPayment = PaymentEntityMockBuilder.forCurrency(CURRENCY.GBP).withAmount(200).build();
+    const firstFeeRecord = firstFeeRecordBuilder.withStatus(FEE_RECORD_STATUS.MATCH).withPayments([matchingPayment]).build();
+    const secondFeeRecord = secondFeeRecordBuilder.withStatus(FEE_RECORD_STATUS.MATCH).withPayments([matchingPayment]).build();
+    cy.task(NODE_TASKS.INSERT_FEE_RECORDS_INTO_DB, [firstFeeRecord, secondFeeRecord]);
+
+    cy.visit(`utilisation-reports/${REPORT_ID}`);
+    pages.utilisationReportPage.premiumPaymentsTab.generateKeyingDataButton().click();
+    pages.checkKeyingDataPage.generateKeyingSheetDataButton().click();
+
+    cy.url().should('eq', relative(`/utilisation-reports/${REPORT_ID}#keying-sheet`));
+
+    pages.utilisationReportPage.keyingSheetTab.keyingSheetTableRow(FIRST_FEE_RECORD_ID).should('contain', 'TO DO');
+    pages.utilisationReportPage.keyingSheetTab.keyingSheetTableRow(SECOND_FEE_RECORD_ID).should('contain', 'TO DO');
+
+    const EXPECTED_PRINCIPAL_BALANCE_ADJUSTMENT = 20000; // 200000 - 180000, DECREASE
+    const EXPECTED_FIXED_FEE_ADJUSTMENT = 7100; // 8100 - 1000, INCREASE
```

Comment:
Please could we include more detail to where the 8100 comes from in case this ever fails and someone less knowledgable about keying sheet generation needs to try and figure out the failure?